demo/server/src/app.ts

```diff
@@ -1,33 +1,112 @@
 'use strict';
 
 import * as express from 'express';
+import * as session from 'express-session';
 import * as path from 'path';
 import * as logger from 'morgan';
 import * as cookieParser from 'cookie-parser';
 import * as bodyParser from 'body-parser';
 import index from './routes/index';
 import * as favicon from 'serve-favicon';
 import EnvironmentConfig, { Config, CloudAppConfig } from './util/config';
+// import * as passport from 'passport';
+// import {Strategy} from 'passport-local';
+import users, {User} from './userSeed';
+import PassportAuthImpl from '@raincatcher/passport-auth';
 
 const app: express.Express = express();
 const appConfig: Config<CloudAppConfig> = new EnvironmentConfig<CloudAppConfig>();
 const config = appConfig.getConfig();
+// const authService = new PassportAuthImpl<User>();
 
 app.use(logger(config.morganOptions));
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
+app.use(session({secret: 'test', resave: false, saveUninitialized: false}));
+authService.init(app);
+// app.use(passport.initialize());
+// app.use(passport.session());
 app.use(express.static(path.join(__dirname, 'public')));
-app.use(favicon(path.join(__dirname, '../public', 'favicon.ico')))
+app.use(favicon(path.join(__dirname, '../public', 'favicon.ico')));
 
 app.use('/', index);
 
 app.use((req: express.Request, res: express.Response, next) => {
-  var err: any = new Error('Not Found');
+  let err: any = new Error('Not Found');
   err.status = 404;
   next(err);
 });
 
+//passport config
+authService.setSerializeFunction(function(user: User, done: Function) {
+  done(null, user.id);
+});
+
+authService.setDeserializeFunction(function(id: string, done: Function) {
+  users.forEach(function(user) {
+    if(user.id === id) {
+      let userObj = {
```

Comment:
I think a +1 to read from the data source rather than from the `req.user` to get the profile data. Best to have the data store be the source of truth, as we can return the user's roles and any other information using the same function.